hledger-lib/Hledger/Data/Types.hs

```diff
@@ -238,11 +238,18 @@ instance Show Status where -- custom show.. bad idea.. don't do it..
 
 data BalanceAssertion = BalanceAssertion {
       baamount   :: Amount,
+      baflags    :: AssertionFlags,
       baposition :: GenericSourcePos
     } deriving (Eq,Typeable,Data,Generic,Show)
 
 instance NFData BalanceAssertion
 
+data AssertionFlags = AssertionFlags {
+      afexact :: Bool
```

Comment:
Let's add some docs, eg -- ^ Does this assertion check the full multi-commodity balance or only one commodity ?

Reply:
Also for BalanceAssertion, eg -- | A statement about the expected account balance after a particular posting.

Reply:
PS I'd be fine with inlining the flags in BalanceAssertion, your call.

Reply:
Good catch.  The `Types` module in general could use better docs, but that's no reason to compound the issue.  And I'm not entirely sure why I wasn't inlining the flags to begin with, thanks!